File: frontend/src/services/roomService.js
import api from "./api";

export const getAllRooms = () => {
    return api.get("/rooms");
};

export const getRoomsByHotel = (hotelId) => {
    return api.get(`/rooms/hotel/${hotelId}`);
};

export const addRoom = (room) => {
    return api.post("/rooms", room);
};

export const updateRoom = (id, room) => {
    return api.put(`/rooms/${id}`, room);
};

export const deleteRoom = (id) => {
    return api.delete(`/rooms/${id}`);
};

File: backend/src/main/java/com/stayease/service/RoomService.java
package com.stayease.service;

import java.util.List;

import com.stayease.dto.request.RoomRequest;
import com.stayease.dto.response.RoomResponse;

public interface RoomService {

    RoomResponse addRoom(RoomRequest request);

    RoomResponse updateRoom(Long roomId, RoomRequest request);

    void deleteRoom(Long roomId);

    List<RoomResponse> getRoomsByHotel(Long hotelId);
    
    List<RoomResponse> getAllRooms();
}